runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -0,0 +1,115 @@
+package org.corfudb.util;
+
+import com.codahale.metrics.CsvReporter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.MetricSet;
+import com.codahale.metrics.jvm.FileDescriptorRatioGauge;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+import com.github.benmanes.caffeine.cache.Cache;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class MetricsUtils {
+    private static final FileDescriptorRatioGauge metricsJVMFdGauge = new FileDescriptorRatioGauge();
+    private static final MetricSet metricsJVMGC = new GarbageCollectorMetricSet();
+    private static final MetricSet metricsJVMMem = new MemoryUsageGaugeSet();
+    private static final MetricSet metricsJVMThread = new ThreadStatesGaugeSet();
+
+    private static Properties metricsProperties = new Properties();
+    private static boolean metricsReportingEnabled = false;
+    private static String mpTrigger = "filter-trigger"; // internal use only
+
+    /**
+     * Load a metrics properties file.
+     * The expected properties in this properties file are:
+     * <p>
+     * enabled: Boolean for whether CSV output will be generated.
+     * For each reporting interval, this function will be
+     * called to re-parse the properties file and to
+     * re-evaluate the value of 'enabled'.  Changes to
+     * any other property in this file will be ignored.
+     * <p>
+     * directory: String for the path to the CSV output subdirectory
+     * <p>
+     * interval: Long for the reporting interval for CSV output
+     */
+    private static void loadPropertiesFile() {
+        String propPath;
+
+        if ((propPath = System.getenv("METRICS_PROPERTIES")) != null) {
+            try {
+                metricsProperties.load(new FileInputStream(propPath));
```

Comment:
I'll add a try with resource to allow the AutoCloseable to do things the correct way.